services/supervisor/src/lib.rs

```diff
@@ -76,6 +81,27 @@ impl Service for Supervisor {
 
             trace!("Removed outdated deployment request {:?}", request);
         }
+
+        let entry = schema.config_propose_with_hash_entry().get();
+        if let Some(entry) = entry {
+            if entry.config_propose.actual_from <= height {
+                // Remove pending config proposal for which deadline was exceeded.
+                trace!("Removed outdated config proposal");
+                schema.config_propose_with_hash_entry().remove();
+            } else {
+                // Apply pending config in case 2/3+1 validators voted for it.
```

Comment:
Nit: I think this comment should be placed before the `if confirmations >= byzantine_quorum(validators) {` line.